### PR TITLE
fragment with bottom nav bar

### PR DESCRIPTION
**Pull request by abhishekRaku**
Fragment and bottom nav bar
Observation:
1. Fragment creation with view building is very easy
2. If you want to have multiple screens in one activity use this.  you can switch between them easily. In rlo project i also saw chatfragment, callfragment. 
3. Use when u need to dynamically add, remove, or replace parts of the UI. 
4. fragment has its own lifecycle helps us to manage the behavior and state of UI components independently. While switching to chatFragment we can load chatList indepently of activity. while Bottom Nav is already there, same with CallFragment.
5. Fragment lifecyle is just little different than activities, same thing: if frament become invisible onPause->onStop
different thing onAttach->OnCreate->OnCreateView -> onViewCreated -> onStart -> onResume

6. we can use directly attach fragment in activity xml ( bad way). Using fragmentManger is better. FragmentTransaction for add,replace, remove fragment operations, commit to commit the operation. We can also use addToBackStack to keep track on Fragments.
7. Other examples: maps fragment inside activity, login fragment, most of the app of tablet we fragment because screen size is big so better user experience it should have multiple screens,

### CODE REVIEW

File: app/src/main/java/com/example/androidlearning/CallFragment.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding = CallFragmentBinding.bind(view)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        binding.clickMeBtnCalls.setOnClickListener {
+            Toast.makeText(context,"This is Call Fragment", Toast.LENGTH_SHORT).show()
+        }
```
Comment (anees-thyrantakath) [inline app/src/main/java/com/example/androidlearning/CallFragment.kt:23]: Why did you define it here? Can you explain?

File: app/src/main/java/com/example/androidlearning/ChatFragment.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding = ChatFragmentBinding.bind(view)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        binding.clickMeBtnChat.setOnClickListener {
+            Toast.makeText(context,"This is Chat Fragment", Toast.LENGTH_SHORT).show()
+        }
```
Comment (anees-thyrantakath) [inline app/src/main/java/com/example/androidlearning/ChatFragment.kt:24]: Same here?

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -6,13 +6,27 @@
...
+    <com.google.android.material.bottomnavigation.BottomNavigationView
+        android:id="@+id/myBottomNav"
+        android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:text="Hello World!"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+        app:menu="@menu/bottom_nav_bar"/>
```
Comment (anees-thyrantakath) [inline app/src/main/res/layout/activity_main.xml:28]: How it's taking care if navigation?

File: app/src/main/java/com/example/androidlearning/CallFragment.kt (hunk 1 above)
Comment (abhishekRaku) [reply app/src/main/java/com/example/androidlearning/CallFragment.kt:23]: Just a button inside fragment class I want to see different button are getting clicked or not in different fragments. That why there are 3 button , 1 button each fragment class.

File: app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (abhishekRaku) [reply app/src/main/res/layout/activity_main.xml:28]: I have add menu resource xml file there, in that xml file I have given 3 menu items. Thus 3 menu are appears in bottom nav bar, then in main activity i'm using setOnItemSelectedListener to get itemId of which menu item user is clicking and then doing operation accordingly.

**Pull request merged**